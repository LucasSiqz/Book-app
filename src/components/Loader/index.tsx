import * as S from './styles'

const Loader = () => (
  <S.LoadingContainer>
    <div className="lds-ellipsis">
      <div />
      <div />
      <div />
      <div />
    </div>
  </S.LoadingContainer>
)

export default Loader
