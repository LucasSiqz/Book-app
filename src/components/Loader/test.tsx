import { render } from 'utils/test-utils'

import Loader from '.'

describe('<Loader />', () => {
  it('should render correctly', () => {
    const { container } = render(<Loader />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
