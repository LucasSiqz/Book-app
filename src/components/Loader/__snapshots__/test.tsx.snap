// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render correctly 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 80px;
  height: 80px;
}

.c0 .lds-ellipsis div {
  position: absolute;
  left: 50%;
  top: 46%;
  -webkit-transform: translate(-50%,-46%);
  -ms-transform: translate(-50%,-46%);
  transform: translate(-50%,-46%);
  width: 13px;
  height: 13px;
  border-radius: 50%;
  background: #FF6978;
  -webkit-animation-timing-function: cubic-bezier(0,1,1,0);
  animation-timing-function: cubic-bezier(0,1,1,0);
}

.c0 .lds-ellipsis div:nth-child(1) {
  left: 8px;
  -webkit-animation: lds-ellipsis1 0.6s infinite;
  animation: lds-ellipsis1 0.6s infinite;
}

.c0 .lds-ellipsis div:nth-child(2) {
  left: 8px;
  -webkit-animation: lds-ellipsis2 0.6s infinite;
  animation: lds-ellipsis2 0.6s infinite;
}

.c0 .lds-ellipsis div:nth-child(3) {
  left: 32px;
  -webkit-animation: lds-ellipsis2 0.6s infinite;
  animation: lds-ellipsis2 0.6s infinite;
}

.c0 .lds-ellipsis div:nth-child(4) {
  left: 56px;
  -webkit-animation: lds-ellipsis3 0.6s infinite;
  animation: lds-ellipsis3 0.6s infinite;
}

<div
  class="c0"
>
  <div
    class="lds-ellipsis"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`
